tests/framework/console/UnkownCommandExceptionTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yiiunit\framework\console;
+
+use Yii;
+use yii\console\Application;
+use yii\console\UnknownCommandException;
+use yiiunit\TestCase;
+
+/**
+ * @group console
+ */
+class UnkownCommandExceptionTest extends TestCase
+{
+    public function setUp()
+    {
+        $this->mockApplication([
+            'enableCoreCommands' => false,
+            'controllerMap' => [
+                'cache' => 'yii\console\controllers\CacheController',
+                'migrate' => 'yii\console\controllers\MigrateController',
+                'message' => 'yii\console\controllers\MessageController',
+            ],
+        ]);
+    }
+
+    public function suggestedCommandsProvider()
+    {
+        return [
+            ['migate', ['migrate']],
+            ['mihate/u', ['migrate/up']],
+            ['mirgte/u', ['migrate/up']],
+            ['mirgte/up', ['migrate/up']],
+            ['mirgte', ['migrate']],
+            ['hlp', ['help']],
+            ['ca', ['cache', 'cache/flush', 'cache/flush-all', 'cache/flush-schema', 'cache/index']],
+            ['cach', ['cache', 'cache/flush', 'cache/flush-all', 'cache/flush-schema', 'cache/index']],
+            ['cach/fush', ['cache/flush']],
+            ['cach/fushall', ['cache/flush-all']],
+            ['what?', []],
+            // test UTF 8 chars
+            ['ёлка', []],
```

Comment:
Is there a reason to test it? Command names are not unicode and the code finding similarities won't work with unicode since it uses non-mb_* string manipulation.